src/mqt/problemsolver/tsp.py

```diff
@@ -212,11 +213,10 @@ def get_all_phases(self) -> list[float]:
         return [a, b, c, d, e, f, g, h, i, j, k, m]
 
     def simulate(self, qc: QuantumCircuit) -> str:
-        backend = DDSIMProvider().get_backend("qasm_simulator")
-        job = execute(qc, backend, shots=1000)
-        count = job.result().get_counts()
-
-        return cast("str", count.most_frequent())
+        qc = qc.decompose().decompose()  # Decompose the circuit to remove any unnecessary gates
```

Comment:
Decompose has a `reps` argument so that you do not need repeated calls: https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.QuantumCircuit#decompose

But these should not really be necessary in the first place. Similar to the other comment.